### ISSUE TITLE
Question about performing limma on raw data

### ISSUE THREAD

**Issue by andrewdchen** [openproblems-bio/task_perturbation_prediction#89]
Hello!

Opening another issue since it's not entirely related to #87, but I was wondering how I might replicate the limma results on the raw data `GSE279945_sc_counts_processed.h5ad` from [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE279945). 

I saw in the Kaggle [description](https://www.kaggle.com/competitions/open-problems-single-cell-perturbations/data) that first a pseudobulk was calculated by averaging profiles of the same cell type within each well. But what I noticed when I looked at the metadata is that there are multiple compounds belonging to the same cell type within each well so it wouldn't be accurate to average these profiles after grouping by cell type and well only?

```
> adata$obs %>% group_by(well, cell_type_orig) %>% summarize(sm_names = paste(unique(sm_name), collapse = "+"), num_sm = n_distinct(sm_name)) %>% dplyr::filter(num_sm == 2)
`summarise()` has grouped output by 'well'. You can override using the `.groups` argument.
# A tibble: 409 × 4
# Groups:   well [69]
   well  cell_type_orig     sm_names                         num_sm
   <fct> <fct>              <chr>                             <int>
 1 A4    B cells            Atorvastatin+Riociguat                2
 2 A4    Myeloid cells      Atorvastatin+Riociguat                2
 3 A4    NK cells           Atorvastatin+Riociguat                2
 4 A4    T cells CD4+       Atorvastatin+Riociguat                2
 5 A4    T cells CD8+       Atorvastatin+Riociguat                2
 6 A4    T regulatory cells Atorvastatin+Riociguat                2
 7 A5    B cells            Ganetespib (STA-9090)+CHIR-99021      2
 8 A5    Myeloid cells      Ganetespib (STA-9090)+CHIR-99021      2
 9 A5    NK cells           Ganetespib (STA-9090)+CHIR-99021      2
10 A5    T cells CD4+       Ganetespib (STA-9090)+CHIR-99021      2
# ℹ 399 more rows
# ℹ Use `print(n = ...)` to see more rows
```

Also, I was wondering if there was a reason averaging is used to find pseudobulks and not the sum?

Any insights would be appreciated!

Thanks,

**Comment by szalata** [comment]
Running this processing pipeline https://github.com/openproblems-bio/task_perturbation_prediction/tree/main/src/workflows/process_dataset on the data downloaded with this script https://github.com/openproblems-bio/task_perturbation_prediction/blob/main/scripts/create_resources/neurips-2023-data.sh would replicate the results.

We are currently updating the code to start from GEO instead of AWS.

Every well contains only a single drug. The confusion comes from the fact that "well" column naming is consistent across plates. You can see the layout in the paper or in the kaggle competition website.

Averaging is used to immediately correct for the variable number of cells in different wells.

**Comment by andrewdchen** [comment]
Gotcha that makes sense! Thanks!

**Comment by andrewdchen** [comment]
It seems like the data in adata.X for both 

`aws s3 cp --no-sign-request \
    s3://openproblems-bio/public/neurips-2023-competition/sc_counts_reannotated_with_counts.h5ad`
and GEO `GSE279945_sc_counts_processed.h5ad` are both log-normalized already (due to the presence of non-integers). 

I think you could still do the pseudobulk of log-normalized data and then do limma but it seems a little unconventional as you may be missing information on the library size (i.e. scaling was done before log-normalization as the manuscript suggests). 

Are the actual raw counts available somewhere? I hope I'm making sense!

Thanks,

**Comment by szalata** [comment]
it is a quirk of anndatas - they have a property `.raw`. That's where you can find the "raw" counts, under anndata.raw.X

**Comment by andrewdchen** [comment]
Hello!

In the process of running the `run_limma` script, I noticed that B Cells treated with Tivozanib appears in both the train set and the public test.

```
                                sm_cell_type cell_type   sm_name sm_lincs_id                                               SMILES       split control
NeurIPS2023-010-A8-B cells Tivozanib_B cells   B cells Tivozanib    LSM-1005 COc1cc2nccc(Oc3ccc(NC(=O)Nc4cc(C)on4)c(Cl)c3)c2cc1OC public_test   FALSE
NeurIPS2023-014-A8-B cells Tivozanib_B cells   B cells Tivozanib    LSM-1005 COc1cc2nccc(Oc3ccc(NC(=O)Nc4cc(C)on4)c(Cl)c3)c2cc1OC       train   FALSE
```

Looking back at the original anndata object from GEO (before pseudobulk), I found that it is true that the same compound treated with the same cell type appears in both (though it's only 1 profile).

```
 adata$obs %>% dplyr::filter(cell_type == "B cells" & sm_name == "Tivozanib") %>% dplyr::pull(split) %>% table
.
     control private_test  public_test        train 
           0            0          230            1 
```

Not sure if I'm missing a pre-processing step (I followed the same `filter_obs` steps in the pipeline), but thought this was worth bringing up!

Thanks,

**Comment by szalata** [comment]
Hello,
There is a difference in processing - note that the splits were updated between the competition and benchmark. Namely, all of the public_test was added to training set (so that no validation set is predefined - every method/user can do their own). only private_test is part of the test set in the released benchmark.

**Comment by szalata** [comment]
seems resolved. Please reopen if needed.
